zendesk/error.go

```diff
@@ -0,0 +1,12 @@
+package zendesk
+
+import "fmt"
+
+// OptionsError is an error type for invalid option argument.
+type OptionsError struct {
```

Comment:
do we really need to export this error type?

Reply:
also given the name of this file it might make sense to move this error type into here.

https://github.com/nukosuke/go-zendesk/blob/master/zendesk/zendesk.go#L27-L56

Reply:
> do we really need to export this error type?

I think it necessary to check error type at application code which uses this library.

```go
if serr, ok := err.(*zendesk.OptionsError); ok {
  // some error handling
}
```

> also given the name of this file it might make sense to move this error type into here.

fixed at fac2798. thanks.